app/models/post.rb

```diff
@@ -0,0 +1,18 @@
+class Post < ApplicationRecord
+  belongs_to :author, class_name: 'User', foreign_key: :author_id
+  has_many :comments
+  has_many :likes
+
+  after_save :update_post_counter
+
+  # Public Methods
+  def five_most_recent_comments
+    comments.order(created_at: :desc).limit(5)
+  end
+
+  private
+
+  def update_post_counter
```

Comment:
[OPTIONAL]

Hi, thank you for submitting your code. I have reviewed it and here are some comments:

- You are using the `update` method with the `count` method. This is inefficient and prone to errors, as it requires an extra database query and does not handle concurrency issues. I kindly suggest that you make use of the increment! method to update the posts counter for a user
- The `increment!` method is a built-in method in Rails that increases the value of an attribute by one and saves the record. It also uses optimistic locking to prevent race conditions. You can read more about it [here](https://apidock.com/rails/ActiveRecord/Base/increment!).

- Here is an example of how your code should look like:

```ruby
# app/models/post.rb
...

def update_posts_counter
  author.increment!(:posts_counter)
end
```

I hope this helps you improve your code. Please let me know if you have any questions or feedback. Happy coding! 😊